### PR TITLE
feat:  sigterm handling in CI runner for log archival

### PR DESCRIPTION
**Pull request by subhashish-devtron**
# Description

Fixes #
added sigterm handling in case of abort on which we need to archive logs as part of cleanup and gracious termination

## How Has This Been Tested?
Tested flow manually and through written automation tests for both AWF and SYSTEM executor

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: App.go [hunk 1]
```diff
@@ -27,33 +27,70 @@ import (
...
 
+	LoggingMode := ""
```
Comment (kripanshdevtron) [inline App.go:56]: Rename it to ``RunMode``

File: Dockerfile [hunk 2]
```diff
@@ -32,8 +32,5 @@ COPY --from=build-env /go/bin/cirunner .
...
 RUN chmod 644 /root/.ssh/config
 
-# entrypoint script with conditional logic
-COPY entrypoint.sh /entrypoint.sh
-RUN chmod +x /entrypoint.sh
-
-ENTRYPOINT ["/entrypoint.sh"]
\ No newline at end of file
+# passing ENABLE_LOGGER as argument to cirunner as default behavior
+ENTRYPOINT ["./cirunner", "ENABLE_LOGGER"]
```
Comment (kripanshdevtron) [inline Dockerfile:36]: ``ENABLE_LOGGER`` rename it to ``PARENT_MODE``

File: App.go [hunk 3]
```diff
@@ -27,33 +27,70 @@ import (
...
+	if len(os.Args) > 1 {
+		LoggingMode = os.Args[1]
+	}
+
+	if os.Getenv(util.InAppLogging) == "true" && LoggingMode == "ENABLE_LOGGER" {
```
Comment (kripanshdevtron) [inline App.go:61]: ``ENABLE_LOGGER`` rename it to ``PARENT_MODE``

File: util/AppLoggingHandler.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	"syscall"
+)
+
+// SpawnProcessWithLogging
+// This method handles the logic for maintaining a local log file for log archival and handles SIGTERM propagation.
+// The subprocess spawned will bypass this function and execute the main logic
+func SpawnProcessWithLogging() {
+
+	// Create an in-memory pipe
+	pr, pw := io.Pipe()
```
Comment (kripanshdevtron) [inline util/AppLoggingHandler.go:18]: will there be any impact because of large log file ??

Comment (subhashish-devtron) [reply util/AppLoggingHandler.go:18]: the rate of data  in and out depends on the amount logs being emitted which is reasonable for our use case, therefore there shouldn't be any impact

**Review by kripanshdevtron — APPROVED** [review]

**Pull request merged**